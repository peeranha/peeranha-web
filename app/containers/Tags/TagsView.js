import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

const TagsView = ({ tags }) => (
  <div>
    {_.orderBy(tags, y => y.popularity, ['desc']).map(x => (
      <div
        style={{ boxShadow: '0 0 2px rgba(0,0,0,0.2)' }}
        className="mb-2 p-2"
        key={x.value}
      >
        <div className="mb-1 d-flex justify-content-between align-items-center">
          <span>{x.label}</span>
        </div>
      </div>
    ))}
  </div>
);

TagsView.propTypes = {
  tags: PropTypes.array.isRequired,
};

export default TagsView;
