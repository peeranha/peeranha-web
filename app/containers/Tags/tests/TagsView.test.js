import TagsView from '../TagsView';

const props = {
  tags: [
    {
      value: 1,
      id: 1,
      popularity: 10,
      label: 'label',
    },
  ],
};

describe('TagsView', () => {
  it('test', () => {
    expect(TagsView(props)).toMatchSnapshot();
  });
});
