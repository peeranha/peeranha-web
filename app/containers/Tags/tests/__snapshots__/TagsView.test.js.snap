// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsView test 1`] = `
<div>
  <div
    className="mb-2 p-2"
    style={
      Object {
        "boxShadow": "0 0 2px rgba(0,0,0,0.2)",
      }
    }
  >
    <div
      className="mb-1 d-flex justify-content-between align-items-center"
    >
      <span>
        label
      </span>
    </div>
  </div>
</div>
`;
